dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventChangeLogService.java

```diff
@@ -55,16 +57,31 @@ Page<EventChangeLog> getEventChangeLog(
   List<TrackedEntityDataValueChangeLog> getTrackedEntityDataValueChangeLogs(
       TrackedEntityDataValueChangeLogQueryParams params);
 
+  @Deprecated(since = "2.42")
   void addTrackedEntityDataValueChangeLog(
       TrackedEntityDataValueChangeLog trackedEntityDataValueChangeLog);
 
+  void addEventChangeLog(
+      Event event,
+      DataElement dataElement,
+      String eventProperty,
+      String currentValue,
+      String previousValue,
+      ChangeLogType changeLogType,
+      String userName);
+
   @Deprecated(since = "2.42")
   int countTrackedEntityDataValueChangeLogs(TrackedEntityDataValueChangeLogQueryParams params);
 
   void deleteTrackedEntityDataValueChangeLog(Event event);
 
+  void deleteEventChangeLog(Event event);
+
   void deleteTrackedEntityDataValueChangeLog(DataElement dataElement);
 
+  @Transactional
```

Comment:
Is this Transactional annotation needed here?

Reply:
nope, it's already in the implementation